test/unit/TokensTest.php

```diff
@@ -0,0 +1,368 @@
+<?php
+
+use Talis\Persona\Client\Tokens;
+
+$appRoot = dirname(dirname(__DIR__));
+if (!defined('APPROOT'))
+{
+    define('APPROOT', $appRoot);
+}
+
+require_once $appRoot . '/test/unit/TestBase.php';
+
+class TokensTest extends TestBase {
+
+    function testEmptyConfigThrowsException(){
+        $this->setExpectedException('InvalidArgumentException',
+            'No config provided to Persona Client'
+        );
+        $personaClient = new Tokens(array());
+    }
+
+    function testNullConfigThrowsException(){
+        $this->setExpectedException('InvalidArgumentException',
+            'No config provided to Persona Client'
+        );
+        $personaClient = new Tokens(null);
+    }
+
+    function testMissingRequiredConfigParamsThrowsException(){
+        $this->setExpectedException('InvalidArgumentException',
+            'Config provided does not contain values for: persona_host,persona_oauth_route,tokencache_redis_host,tokencache_redis_port,tokencache_redis_db'
+        );
+        $personaClient = new Tokens(array(
+            'persona_host' => null,
+            'persona_oauth_route' => null,
+            'tokencache_redis_host' => null,
+            'tokencache_redis_port' => null,
+            'tokencache_redis_db' => null,
```

Comment:
I guess it's okay to have trailing commas or this wouldn't work...looks odd though